Feature/12 users can create posts in groups

chenaicirt opened the pull request:
Created POST/GET routes for three content types: post, group, comment. 

From Trello tickets:

https://trello.com/c/1vViM3eU/25-create-post-api-route
https://trello.com/c/km4syoWS/32-create-group-api-route
https://trello.com/c/J8dJnKb9/33-create-comment-api-route

chenaicirt committed:
edited comment controllers and routes for post command, need to update post and group posts as well

chenaicirt committed:
created post routes and controller to be able to create post

chenaicirt committed:
create new post functionality/route works

chenaicirt committed:
post get method for id works

chenaicirt committed:
Added lib to access getPost method

chenaicirt committed:
created create comment route

chenaicirt committed:
added author field

chenaicirt committed:
edited/looked into groups route

chenaicirt committed:
cleaned up comment controller

dkim1546 committed:
Fixed comment routes

zaclittleberry commented on lib/comment/comment.controllers.js:
```diff
@@ -0,0 +1,141 @@
+    },
+
+    getComment: async function(request, h) {
+      const returnComment = getComment(request.params.id);
```
Ideally getComment would be wrapped in a `try` `catch` block. Also, ideally database calls wouldn't have their own `.then` `.catch` functions, because they should catch in the try catch blocks (so we can respond appropriately with a `throw`). (I know, you almost certainly based this off of code where I did this. I didn't realize this was sometimes problematic until later in the project.).

zaclittleberry reviewed (commented):
I just have one suggestion for using a wrapping try/catch in a place where you didn't.

There's a lot of changes in here that are a part of another pull request. I'm going to wait to merge this until the requested changes are made in the other PR and is merged. You may need to/ should merge develop back into your branch, resolve any conflicts (most likely taking changes from develop), and re-push the code back up to your branch. I can help with this if you need.

chenaicirt committed:
merged PR#6 with PR#7

chenaicirt committed:
"another merge"
Merge branch 'feature/demo-adding-post-entity' into feature/12-users-can-create-posts-in-groups

chenaicirt committed:
added try/catch on getComment similar to how getComments has one

zaclittleberry commented on lib/comment/comment.controllers.js:
```diff
@@ -0,0 +1,146 @@
+    getComment: async function(request, h) {
+      let returnComment;
+      try {
+        const returnComment = getComment(request.params.id);
```
shouldn't instantiate with `const` after already instantiating with `let`.

zaclittleberry replied:
also need to `await` getComment

zaclittleberry commented on lib/comment/comment.routes.js:
```diff
@@ -0,0 +1,58 @@
+          auth: 'jwt',
+          description: 'Get Comments',
+          notes: 'Returns all comments.',
+          tags: ['api', 'Comment'],
```
All tags but `api` should be plural.

zaclittleberry commented on lib/comment/comment.routes.js:
```diff
@@ -0,0 +1,58 @@
+          auth: 'jwt',
+          description: 'Get Comment by id',
+          notes: 'Returns one comment.',
+          tags: ['api', 'Comment'],
```
see above. pluralize

zaclittleberry commented on lib/comment/comment.routes.js:
```diff
@@ -0,0 +1,58 @@
+         handler: controllers.postComment,
+         description: 'Create a Comment For Post',
+         notes: 'Create a comment in the database.',
+         tags: ['api', 'Comment', 'Post', 'Group'],
```
see above. pluralize

zaclittleberry committed:
Merge branch 'develop' into feature/12-users-can-create-posts-in-groups

zaclittleberry commented on lib/comment/comment.routes.js:
```diff
@@ -1,32 +1,32 @@
+          tags: ['api', 'Comment'],
+          validate: {
+            query: commentModels.apiFilterQuery,
+          }
```
this should be commented out for mvp/until it is actually implemented based on user story.

zaclittleberry commented on lib/post/post.routes.js:
```diff
@@ -3,31 +3,42 @@ const Joi = require('joi');
+         handler: controllers.postPost,
+         description: 'Create Post',
+         notes: 'Create a Post in the Specified Group.',
+         tags: ['api', 'Post', 'Group'],
```
Pluralize tags except api

zaclittleberry commented on lib/post/post.routes.js:
```diff
@@ -3,31 +3,42 @@ const Joi = require('joi');
+         auth: 'jwt',
+         description: 'Get Post from Post Id',
+         notes: 'Returns a post',
+         tags: ['api', 'Post'],
```
pluralize tags except api

zaclittleberry commented on lib/post/post.controllers.js:
```diff
@@ -0,0 +1,157 @@
+      }
+
+      if (groupItem === null) {
+        throw Boom.badRequest('postItem does not exist');
```
correct error message to say Group (not post)

zaclittleberry commented on lib/post/post.controllers.js:
```diff
@@ -0,0 +1,157 @@
+
+      let groupItem;
+      try {
+        groupItem = await getGroupItem(groupId);
```
this is overcomplicated in the same way as getPostItem. Please correct in similar fashion.

zaclittleberry reviewed (commented):
Ok. getComment is the only thing broken I think. But there are a few other things to change ideally.

chenaicirt committed:
pluralized the api tags

chenaicirt committed:
simplified getPostItem getGroupItem and then also fixed getComment

chenaicirt committed:
Merge branch 'feature/12-users-can-create-posts-in-groups' of https://github.com/GazeDev/groups_api into feature/12-users-can-create-posts-in-groups

zaclittleberry reviewed (approved):
Sorry, for my instructions to 'simplify' the use of getPostItem and getGroupItem I thought I had posted a comment with code, but it doesn't seem I did. What I was actually suggesting was to get rid of 'getPostItem' and 'getGroupItem' functions altogether, and instead where they were called, make the call out to the postLib and groupLib methods instead. What you have will work, it's just repeating code that should be 'owned' by other modules.
